### ISSUE TITLE
Lr schedulers should be called either every step or when evaluating

### ISSUE THREAD

**Issue by bastings** [joeynmt/joeynmt#18]
Certain learning rate schedulers expect to be called at every training step, such as ExponentialLR.
Others expect to be called at every evaluation with the evaluation result. 
There is currently no logic for this distinction.

**Comment by juliakreutzer** [comment]
Fixed in PR #27.
